modules/git/repo_branch_gogit.go

```diff
@@ -101,9 +102,42 @@ func WalkReferences(ctx context.Context, repoPath string, walkfn func(string) er
 	defer iter.Close()
 
 	err = iter.ForEach(func(ref *plumbing.Reference) error {
-		err := walkfn(string(ref.Name()))
+		err := walkfn(ref.Hash().String(), string(ref.Name()))
 		i++
 		return err
 	})
 	return i, err
 }
+
+// WalkReferences walks all the references from the repository
+func (repo *Repository) WalkReferences(arg string, skip, limit int, walkfn func(sha1, refname string) error) (int, error) {
```

Comment:
one concern, I would rather have some "enum" const than arbitrary string

Reply:
Yes, I didn't realise when I wrote this that I was leaking the --tags out of git modules.

I think we already have an appropriate type that we can use here.